### PR TITLE
Set parent of `expr_env` to parent of `check_env` in `grade_this()`

### PR DESCRIPTION
**Pull request by rossellhayes**

### CODE REVIEW

**Review by rossellhayes — COMMENTED** [review]
@gadenbuie Your changes look great!

File: tests/testthat/test-grade_this.R [hunk 1]
```diff
@@ -17,3 +17,29 @@ test_that("grade_this() can return early", {
...
+      from_parent = from_parent,
+      from_setup = from_setup,
+      from_user = .envir_result$from_user
+    ))
+  )(ex)$data
+
+  expect_true(found$from_parent)
+  expect_true(found$from_setup)
+  expect_true(found$from_user)
+})
```
Comment (rossellhayes) [inline tests/testthat/test-grade_this.R:45]: This is great, I was planning to add tests within `tblcheck` once this merged, but it's even nicer to have it all handled here!

Comment (gadenbuie) [reply tests/testthat/test-grade_this.R:45]: Thanks! Hopefully it'll be an early warning system

**Review by gadenbuie — APPROVED** [review]
Looks great, go team!

**Pull request merged**